build.rs

```diff
@@ -227,6 +232,11 @@ fn has_small_stack(_: &Target) -> Result<bool, Box<dyn Error>> {
 }
 
 fn setup_paths() {
+    #[cfg(unix)]
+    use std::path::PathBuf;
+    #[cfg(windows)]
+    use unix_path::{Path, PathBuf};
```

Comment:
What's the need for unix_path here?

Reply:
When cross compiling from windows, the paths need to keep unix style.

Reply:
In what way? "/" vs "\"?

If this really is needed, the dependency should be made windows-specific in Cargo.toml.

Reply:
The original code tries to get the full path from the env vars, making the env vars containing "C:\" even "D:\" depending on the host machine.

I'll make the dep windows only later.

Reply:
I just realized that it's impossible to make `unix-path` windows-specific, because `[target.'cfg(windows)']` means the *target* is Windows, but not *host* is Windows. There's no way to specify a Windows-host only build dependency.

Reply:
I would be fine with just panic!ing if the passed paths start with "X:".

These aren't paths we get from *the system*, these are paths we're passed by *the build configuration*. It should just give them as unix versions.

Reply:
Yes, they are passed by the *build configuration*, but the env var `PREFIX` should be *absolute* according to the code below, and then all relative env vars will be set to absolute path according to `PREFIX`. If you just panic if the passed paths start with drive letter, but also panic if `PREFIX` is not absolute, it would be nearly impossible to compile for cygwin, because the most possible way to compile for cygwin is to cross-compile from Windows.
`unix-path` is used here to minimize and simplify the change of code. Could you explain why you would not like a build dependency that's even not compiled into the final binary?

Reply:
> the most possible way to compile for cygwin is to cross-compile from Windows.

Wouldn't you do something like `cmake -DPREFIX=/usr` or `PREFIX=/usr cargo build`?

> Could you explain why you would not like a build dependency that's even not compiled into the final binary?

Because it would make everyone on linux/macos/bsd and especially our packagers (debian wants to have one source package per rust dependency) install a dependency that they would never need.

If we absolutely have to okay, but I'd like to find a solution that doesn't impact everyone not using cygwin.

Reply:
I'm afraid we have to use it. `/usr` is not an absolute path on Windows, and `\` is not a valid path separator on Unix. Maybe someday there could be a cygwin-hosted rustc, and then we would be able to get rid of this dependency, but it's far from now.

Another possible solution is to create a new crate to handle the paths, which could decide dependency based on platform. It could be a stub crate that re-exports `std::path::*` on Unix and re-exports `unix_path::*` on Windows. We can reference it as a build-dependency and the packagers might be happier about that. (Might be, because anyway `unix_path` will be in the lock file.)

Reply:
Oh, there's another dirtier solution: make `unix_path` as cygwin target only build dependency.

Reply:
What's dirtier about that?

Reply:
We wish to find sth like `[host.'cfg(windows)'.build-dependencies]`, and it's roughly equivalent to `[target.'cfg(target_os = "cygwin")'.build-dependencies]`, because only cygwin target is cross-compiled from Windows. The dirty part is that it's actually possible to cross-compile to cygwin from Linux (I have never tried, but someone did). If you prefer this solution, it's OK to me.

Reply:
If you cross-compiled to cygwin from linux, wouldn't you have unix paths to begin with?

But yeah, that would sound alright.

Reply:
OK, I have set `unix_path` as cygwin only.

Reply:
Oh, maybe it's my fault. `[target.'cfg(windows)'.build-dependencies]` just works fine now...